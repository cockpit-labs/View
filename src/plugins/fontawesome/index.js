import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faCaretDown,
  faCaretLeft,
  faCaretRight,
  faAngleLeft,
  faAngleRight,
  faChevronDown,
  faAsterisk,
  faTimes,
  faMinus,
  faPlus,
  faCircleNotch,
  faChevronLeft,
  faComment,
  faCamera,
  faPen,
  faTrash,
  faExclamation,
  faBell,
  faInfo,
  faCheckSquare,
  faFilePdf

} from '@fortawesome/free-solid-svg-icons'

library.add(
  faCaretDown,
  faCaretLeft,
  faCaretRight,
  faAngleLeft,
  faAngleRight,
  faChevronDown,
  faAsterisk,
  faTimes,
  faMinus,
  faPlus,
  faCircleNotch,
  faChevronLeft,
  faComment,
  faCamera,
  faPen,
  faTrash,
  faExclamation,
  faBell,
  faInfo,
  faCheckSquare,
  faFilePdf
)
