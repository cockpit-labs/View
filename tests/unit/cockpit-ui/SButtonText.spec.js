import { shallowMount } from '@vue/test-utils'
import '@/plugins/fontawesome'
import SButtonText from '@/plugins/cockpit-ui/components/button/SButtonText'

describe('Button', () => {
  const wrapper = shallowMount(SButtonText, {
    propsData: {
      icon: 'times'
    },
    slots: {
      default: 'Hello world'
    }
  })

  it('should contain label', () => {
    const button = wrapper.find('button')
    expect(button.text()).toContain('Hello world')
  })

  it('should contain an icon', () => {
    const icon = wrapper.find('sicon-stub')
    expect(icon.attributes('name')).toBe('times')
  })

  it('should emit an event click on click', () => {
    const button = wrapper.find('button')
    const spy = jest.fn()
    wrapper.vm.$on('click', spy)
    button.trigger('click')
    expect(spy).toHaveBeenCalledTimes(1)
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
