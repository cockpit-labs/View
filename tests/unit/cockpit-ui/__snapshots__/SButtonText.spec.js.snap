// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<button
  class="s-button-text"
>
  <sicon-stub
    fw="true"
    name="times"
  />
   Hello world
</button>
`;
